src/dbnode/persist/fs/commitlog/reader.go

```diff
@@ -92,49 +94,38 @@ type decoderArg struct {
 	bufPool              chan []byte
 }
 
-type readerMetadata struct {
+type completedDecoders struct {
 	sync.RWMutex
-	numBlockedOrFinishedDecoders int64
+	numCompletedDecoders int64
+	wg                   *sync.WaitGroup
 }
 
 type reader struct {
+	// The commit log reader is not goroutine safe, but we use an embedded
+	// mutex to defensively protect the Open/Close methods against misuse.
+	// We don't protect the Read() method with the mutex because it would
+	// be too expensive.
+	sync.Mutex
```

Comment:
Could you perhaps use a sub-struct for this perhaps?

Reply:
Also nit, we use `closed` everywhere so to be consistent perhaps use that?  And if you're truly against that, removing the `is...` prefix would be good as we try to avoid that.